src/main/java/org/team1540/robot2018/Robot.java

```diff
@@ -376,4 +204,199 @@ public void teleopPeriodic() {
+        new DecisionNode(Posession.SWITCH_OWNED_RIGHT, MotionProfile.MIDDLE_TO_RIGHT_SWITCH),
+        new DecisionNode(Posession.NO_DATA, MotionProfile.GO_STRAIGHT).setMessage(new Message("Could not get match data, reverting to base auto", true))
+    })),
+    CENTER_DOUBLE_CUBE("Center Double Cube", new DecisionNode(new DecisionNode[]{
+       new DecisionNode(Posession.SWITCH_OWNED_LEFT, MotionProfile.SWITCH_DOUBLE_CUBE_LEFT),
+       new DecisionNode(Posession.SWITCH_OWNED_RIGHT, MotionProfile.SWITCH_DOUBLE_CUBE_RIGHT)
+    })),
+    RIGHT_HOOK_SWITCH("Right Hook Switch", new DecisionNode(new DecisionNode[]{
+        new DecisionNode(Posession.SWITCH_OWNED_RIGHT, MotionProfile.RIGHT_HOOK),
+        new DecisionNode(Posession.NO_DATA, MotionProfile.GO_STRAIGHT)
+    })),
+    RIGHT_SCALE_THEN_SWITCH("Right Scale Then Switch", new DecisionNode(new DecisionNode[]{
+        new DecisionNode(Posession.SCALE_OWNED_RIGHT, MotionProfile.RIGHT_SCALE),
+        new DecisionNode(Posession.SCALE_OWNED_LEFT, new DecisionNode[]{
+            new DecisionNode(Posession.SCALE_OWNED_RIGHT, MotionProfile.RIGHT_HOOK),
+            new DecisionNode(Posession.SWITCH_OWNED_LEFT, MotionProfile.GO_STRAIGHT),
+            new DecisionNode(Posession.NO_DATA, MotionProfile.GO_STRAIGHT).setMessage(new Message("Could not get match data, reverting to base auto", true))
+        }),
+        new DecisionNode(Posession.NO_DATA, MotionProfile.GO_STRAIGHT).setMessage(new Message("Could not get match data, reverting to base auto", true))
+    })),
+    CROSS_LINE("Cross Line", new DecisionNode(MotionProfile.GO_STRAIGHT)),
+    STUPID("Stupid", new DecisionNode(MotionProfile.DRIVE_TIMED));
+
+
+    private final String name;
+    private final DecisionNode root;
+
+    AutoMode(String name, DecisionNode root) {
+      this.name = name;
+      this.root = root;
+    }
+
+  }
+
+  private enum Posession {
+
+    SCALE_OWNED_LEFT(() -> (MatchData.getOwnedSide(GameFeature.SCALE) == OwnedSide.LEFT)),
+    SCALE_OWNED_RIGHT(() -> (MatchData.getOwnedSide(GameFeature.SCALE) == OwnedSide.RIGHT)),
+    SWITCH_OWNED_LEFT(() -> (MatchData.getOwnedSide(GameFeature.SWITCH_NEAR) == OwnedSide.LEFT)),
+    SWITCH_OWNED_RIGHT(() -> (MatchData.getOwnedSide(GameFeature.SWITCH_NEAR) == OwnedSide.RIGHT)),
+    NO_DATA(() -> (true));
+
+    @NotNull
+    private final BooleanSupplier condition;
+    @Nullable
+    private final String message;
+    private final boolean isError;
+
+    Posession(@NotNull BooleanSupplier condition) {
+      this.condition = condition;
+      this.message = null;
+      isError = false;
+    }
+
+    Posession(@NotNull BooleanSupplier condition, @Nullable String message, boolean isError) {
+      this.condition = condition;
+      this.message = message;
+      this.isError = isError;
+    }
+
+  }
+
+  private enum MotionProfile {
+    GO_STRAIGHT(new SimpleProfileAuto("go_straight"), "Crossing the line"),
```

Comment:
It should be pretty easy to just construct a message using the name of the enum class.

Reply:
?

Reply:
Instead of needing the message to be defined just make it always `"Running routine " + profile.toString()`

Reply:
Done. You didn't write it that way tho :|